drivers/serial/uart_stm32.c

```diff
@@ -675,6 +675,29 @@ static int uart_stm32_err_check(const struct device *dev)
 	return err;
 }
 
+#ifdef CONFIG_UART_WIDE_DATA
+
+static void uart_stm32_poll_out_u9(const struct uart_stm32_config *config, void *out)
+{
+	LL_USART_TransmitData9(config->usart, *((uint16_t *)out));
+}
+
+static void uart_stm32_poll_in_u9(const struct uart_stm32_config *config, void *in)
+{
+	*((uint16_t *)in) = LL_USART_ReceiveData9(config->usart);
+}
+
+static int uart_stm32_poll_in_u16(const struct device *dev, uint16_t *p_u16)
```

Comment:
I still think the function naming is confusing:
`uart_stm32_poll_in_u16/uart_stm32_poll_in` are 'API functions' while `uart_stm32_poll_in_u8/uart_stm32_poll_in_u9` are arguments to the visitor function. (same for poll_out ... ) of course.
I suggest not using `uart_stm32` prefix for arg functions (avoid `__` also which usage is restricted)

Reply:
Agreed, will change